context/app/static/js/components/Detail/provenance/ProvVis/ProvData.js

```diff
@@ -126,7 +127,10 @@ export default class ProvData {
     return Object.entries(this.prov.activity).reduce((acc, [activityId, activity]) => {
       // activity names currently vary across environments
       // TODO reduce to single value once apis are consistent
-      if (!['Register Donor Activity', 'Create Donor Activity'].includes(activity['hubmap:creation_action'])) {
+      if (
+        !['Register Donor Activity', 'Create Donor Activity'].includes(activity['hubmap:creation_action']) ||
+        this.entity_type === 'Donor'
```

Comment:
Can you move the comment above inside the `if`, and then add a new comment about `this.entity_type === 'Donor'` ... Right now, it reads as if the comment applies to everything.